va/va_dec_av1.h

```diff
@@ -1,656 +1,660 @@
+    uint8_t                 num_y_points;
+    uint8_t                 point_y_value[14];
+    uint8_t                 point_y_scaling[14];
+    /*  value range [0..10] */
+    uint8_t                 num_cb_points;
+    uint8_t                 point_cb_value[10];
+    uint8_t                 point_cb_scaling[10];
+    /*  value range [0..10] */
+    uint8_t                 num_cr_points;
+    uint8_t                 point_cr_value[10];
+    uint8_t                 point_cr_scaling[10];
+    /*  value range [-128..127] */
+    int8_t                  ar_coeffs_y[24];
+    int8_t                  ar_coeffs_cb[25];
+    int8_t                  ar_coeffs_cr[25];
+    uint8_t                 cb_mult;
+    uint8_t                 cb_luma_mult;
+    uint16_t                cb_offset;
+    uint8_t                 cr_mult;
+    uint8_t                 cr_luma_mult;
+    uint16_t                cr_offset;
+
+    /** \brief Reserved bytes for future use, must be zero */
+    uint32_t                va_reserved[VA_PADDING_LOW];
+
+} VAFilmGrainStructAV1;
+
+typedef enum {
+    /** identity transformation, 0-parameter */
+    VAAV1TransformationIdentity           = 0,
+    /** translational motion, 2-parameter */
+    VAAV1TransformationTranslation        = 1,
+    /** simplified affine with rotation + zoom only, 4-parameter */
+    VAAV1TransformationRotzoom            = 2,
+    /** affine, 6-parameter */
+    VAAV1TransformationAffine             = 3,
+    /** transformation count */
+    VAAV1TransformationCount
+} VAAV1TransformationType;
+
+typedef struct _VAWarpedMotionParamsAV1{
+
+    /** \brief Specify the type of warped motion */
+    VAAV1TransformationType  wmtype;
+
+    /** \brief Specify warp motion parameters
+     *  wm.wmmat[] corresponds to gm_params[][] in spec.
+     *  Details in AV1 spec section 5.9.24 or refer to libaom code
+     *  https://aomedia.googlesource.com/aom/+/refs/heads/master/av1/decoder/decodeframe.c
+     */
+    int32_t                 wmmat[8];
+
+    /* valid or invalid on affine set */
+    uint8_t  invalid;
+
+    /** \brief Reserved bytes for future use, must be zero */
+    uint32_t                va_reserved[VA_PADDING_LOW];
+
+} VAWarpedMotionParamsAV1;
+
+/**
+ * \brief AV1 Decoding Picture Parameter Buffer Structure
+ *
+ * This structure conveys picture level parameters and should be sent once
+ * per frame.
+ *
+ */
+typedef struct  _VADecPictureParameterBufferAV1
+{
+    /**@{*/
+
+    /** \brief sequence level information
+     */
+
+    /** \brief AV1 bit stream profile
+     */
+    uint8_t                 profile;
+
+    uint8_t                 order_hint_bits_minus_1;
+
+    /** \brief bit depth index
+     *  value range [0..2]
+     *  0 - bit depth 8;
+     *  1 - bit depth 10;
+     *  2 - bit depth 12;
+     */
+    uint8_t                 bit_depth_idx;
+
+    /** \brief corresponds to AV1 spec variable of the same name. */
+    uint8_t                 matrix_coefficients;
```

Comment:
Hi @XinfengZhang , @uartie , am I right that this is now a backward compatible change? I mean adding "uint8_t matrix_coefficients" to the middle of the structure? Will we have to rebuild all libva based applications(ffmpeg, msdk, etc)?

Reply:
This header file was just merged yesterday.  So I doubt applications are using them already.  Either way, it hasn't been included in any release, as of yet.

Reply:
it will not break compatible , but the variable name change will lead to compatible issue, but it just merged 15 hours ago, so we have chance to change it

Reply:
@uartie @XinfengZhang this insertion will not break compatibility only if compiler applies reasonable padding policy. Much safer is to insert new fields to the end of structs or reserve space in the middle with "reserved" fields, which can be replaced by some meaningful value in future releases. But anyway you cannot guarantee binary compatibility of structures layout without fixing it with special compiler pragmas. Even update of compiler without code change may result in changed structures binary layout. You can [look at our MSDK approach](https://github.com/Intel-Media-SDK/MediaSDK/pull/1426) which guarantees desired binary layout of our API structs. The only way to check that binary layout stays the same is fixing offset of each struct's field to precise value in ```static_assert``` like we did [here](https://github.com/Intel-Media-SDK/MediaSDK/blob/master/_studio/shared/src/mfx_static_assert_structs.cpp)

Reply:
Folks, I understand that this was not released, but, please, let's try to avoid such things in a future. I think it's much better and developer friendly to have master branch always supported in a backward compatible style. Otherwise developers may start to see random failures just because they forgot to update libva on their system! I really would like to limit backward incompatible changes in master branch to the absolutely required show stopper things. Everything else means that we did not spend enough time doing code review.

Reply:
Yes, it's unfortunate that the initial code review for this public API was not thorough enough before it got merged.  I don't think anyone is trying to intentionally make a habit out of doing postmortem changes like this when that happens.  And, considering if things left unsaid, I think everyone already understands the consequences of doing such things.

The [time] window between the initial API merge and these changes was very small, so the call was made to allow it based on that.  I doubt there was any harm done other than slightly inconveniencing or annoying a few people.

Either way, this proves that it is very important to thoroughly review new public API's before they are merged.  The more eyes, the better.  And **coding standards really should be put in place and enforced** to help with this matter.  Otherwise, we just end up with sloppy and inconsistent interfaces that only discourage people from using them.

If the initial commit did not break my compilation due to missing install header then, unfortunately, I probably wouldn't have noticed some of the other problems I reported (i.e. code style and DOS format).